test/sshkit/ssh_test.exs

```diff
@@ -13,36 +13,150 @@ defmodule SSHKit.SSHTest do
 
   defmodule SSHSandboxError do
     def connect(_, _, _, _), do: {:error, :sandbox}
+
+    def close(_) do
+      send self(), :closed_sandbox_connection
+      :ok
+    end
+  end
+
+  defmodule SSHSandboxConnectionSuccess do
+    def session_channel(:sandbox, _, _, _), do: {:ok, 0}
+
+    def exec(_, _, _, _) do
+      send self(), :exec_sandbox_connection
+      {:ok, :result}
+    end
+  end
+
+  defmodule SSHSandboxConnectionError do
+    def session_channel(:sandbox, _, _, _), do: {:ok, 0}
+
+    def exec(_, _, _, _) do
+      send self(), :exec_sandbox_connection
+      :failure
+    end
+  end
+
+  setup_all do
+    {:ok, host: "foo.io"}
+  end
+
+  describe "connect/2" do
+    @options [ssh_modules: %{ssh: SSHSandboxSuccess, ssh_connection: :ssh_connection}]
+    test "open sandbox connection with given options and keep it open", state do
+      user = "test"
+      options = @options ++ [user: user]
+      conn = %SSHKit.SSH.Connection{
+        host:        String.to_charlist(state[:host]),
```

Comment:
you could "charlist'ify" the how in the setup-block (or module variable definition) to make this line more readable